File: compiler/testData/foreignAnnotationsJava8/tests/jspecify/java/IgnoreAnnotations.java
import org.jspecify.annotations.*;

public class IgnoreAnnotations {
    public static class Base {
        void foo() {}
    }
    public static class Derived extends Base { }

    @Nullable public Derived field = null;

    @Nullable
    public Derived foo(Derived x, @NullnessUnspecified Base y) {
        return null;
    }

    static void main(IgnoreAnnotations a) {
        a.foo(new Derived(), null).foo();
        a.foo(null, new Derived()).foo();

        a.field.foo();
    }
}

File: compiler/testData/foreignAnnotationsJava8/tests/jspecify/ignoreAnnotations.kt
// !DIAGNOSTICS: -UNUSED_VARIABLE -UNUSED_PARAMETER
// CODE_ANALYSIS_STATE ignore
// FILE: A.java

import org.jspecify.annotations.*;

public class A {
    @Nullable public String field = null;

    @Nullable
    public String foo(String x, @NullnessUnspecified CharSequence y) {
        return "";
    }
}

// FILE: main.kt

fun main(a: A) {
    a.foo("", null)?.length
    a.foo("", null).length
    a.foo(null, "").length

    a.field?.length
    a.field.length
}
